exporters/zipkin/src/main/java/io/opentelemetry/exporter/zipkin/EventDataToAnnotation.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.exporter.zipkin;
+
+import static java.util.stream.Collectors.joining;
+
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.sdk.trace.data.EventData;
+import java.util.List;
+
+/**
+ * Converts an EventData instance to a String representation of that data, with attributes converted
+ * to JSON.
+ *
+ * <p>See <a
+ * href="https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/sdk_exporters/zipkin.md#events">the
+ * zipkin exporter spec</a> for details.
+ */
+class EventDataToAnnotation {
+
+  static String apply(EventData eventData) {
+    String name = eventData.getName();
+    String value = toJson(eventData.getAttributes());
+    return "\"" + name + "\":" + value;
+  }
+
+  private static String toJson(Attributes attributes) {
+    return attributes.asMap().entrySet().stream()
+        .map(entry -> "\"" + entry.getKey() + "\":" + toValue(entry.getValue()))
```

Comment:
WDYT about using a single `StringBuilder` instead of glueing multiple strings together?

Reply:
It felt unnecessary here, and this is one of the very few optimizations the compiler will actually do for us (I confirmed it in the bytecode too). I think it reads better this way.